src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -0,0 +1,57 @@
+package com.rocketden.main.service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.rocketden.main.dao.RoomRepository;
+import com.rocketden.main.dto.game.GameDto;
+import com.rocketden.main.dto.notification.NotificationDto;
+import com.rocketden.main.model.Game;
+import com.rocketden.main.model.PlayerCode;
+import com.rocketden.main.model.Room;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public abstract class GameManagementService {
+
+    private final RoomRepository repository;
+    private final SocketService socketService;
+    private Map<String, Game> currentGameMap;
+
+    @Autowired
+    protected GameManagementService(RoomRepository repository, SocketService socketService) {
+        this.repository = repository;
+        this.socketService = socketService;
+        currentGameMap = new HashMap<>();
+    }
+
+    public Game createGameFromRoom(Room room) {
+        // TODO: Create the game from the room (or roomId).
+        return null;
+    }
+
+    public void addGame(Game game) {
+        currentGameMap.put(game.getRoomId(), game);
+    }
+
+    protected Game getGameFromRoomId(String roomId) {
+        return currentGameMap.get(roomId);
+    }
+
+    protected void removeGame(String roomId) {
+        currentGameMap.remove(roomId);
+    }
```

Comment:
I thought this was the basis for why we wanted this class? So that these methods could handle the game management, and connect to the methods below (notifications, submissions, etc.)?

Reply:
Yeah that's fair - I think these methods can also deal with error checking like if the roomId doesn't exist, so that can help reduce some duplicate code